pkgs/applications/blockchains/snarkvm/default.nix

```
@@ -0,0 +1,48 @@
+{
+  fetchFromGitHub,
+  lib,
+  rustPlatform,
+  pkg-config,
+  openssl,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "snarkVM";
+  version = "1.2.1";
+
+  src = fetchFromGitHub {
+    owner = "ProvableHQ";
+    repo = pname;
+    rev = "/refs/heads/mainnet";
+    sha256 = "sha256-XpadXcWKa3Eev/BjmNCQLlQlYkniDgge5CxDUBBDSeU=";
+  };
+
+  cargoHash = "sha256-6V/GdkdfzBWJ1gFcvvWz3WfFD/HWZcQnSN0S5D5XAGw=";
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ openssl ];
+
+  checkPhase = ''
+    cargo test -- --skip=package::tests::test_package_run_and_execute_match \
+      --skip=file::prover::tests::test_create_and_open \
+      --skip=file::verifier::tests::test_create_and_open \
+      --skip=package::is_build_required::tests::test_prebuilt_package_does_not_rebuild \
+      --skip=package::clean::tests::test_clean \
+      --skip=package::run::tests::test_run \
+      --skip=package::build::tests::test_build \
+      --skip=package::deploy::tests::test_deploy \
+      --skip=package::run::tests::test_run_with_nested_imports \
+      --skip=package::deploy::tests::test_deploy_with_import \
+      --skip=package::clean::tests::test_clean_with_import \
+      --skip=package::run::tests::test_run_with_import \
+      --skip=package::build::tests::test_build_with_import
+  '';
+
+  meta = with lib; {
+    description = "zero-knowledge virtual machine (zk-VM)";
```

Proper capitalization
```suggestion
    description = "Zero-knowledge virtual machine";
```